nltk/stem/arlstem.py

```diff
@@ -17,354 +17,409 @@
+        
+    def suff(self, token):
+        """ 
+            remove suffixes from the word's end 
+        """
+        if(token.endswith('\u0643') and len(token)>3):
+            token = token[:-1]
+            return token
+        if(len(token) > 4):
+            for s2 in self.su2:
+                if(token.endswith(s2)):
+                    token = token[:-2]
+                    return token
+        if(len(token) > 5):
+            for s3 in self.su3:
+                if(token.endswith(s3)):
+                    token = token[:-3]
+                    return token
+        if(token.endswith('\u0647') and len(token)>3):
+            token = token[:-1]
+            return token        
+        if(len(token) > 4):
+            for s2 in self.su22:
+                if(token.endswith(s2)):
+                    token = token[:-2]
+                    return token
+        if(len(token) > 5):
+            for s3 in self.su32:
+                if(token.endswith(s3)):
+                    token = token[:-3]
+                    return token
+        if(token.endswith('\u0646\u0627') and len(token)>4):
+            token = token[:-2]
+            return token                        
+        return token
+        
+    def fem2masc(self, token):
+        """ 
+            transform the word from the feminine form to the masculine form 
+        """
+        if(token.endswith('\u0629') and len(token)>3):
+            token = token[:-1]
+            return token
+        
+    def plur2sing(self, token):
+        """ 
+            transform the word from the plural form to the singular form 
+        """
+        if(len(token) > 4):
+            for ps2 in self.pl_si2:
+                if(token.endswith(ps2)):
+                    token = token[:-2]
+                    return token
+        if(len(token) > 5):
+            for ps3 in self.pl_si3:
+                if(token.endswith(ps3)):
+                    token = token[:-3]
+                    return token
+        if(len(token)>3 and token.endswith('\u0627\u062A')):
+            token = token[:-2]
+            return token
+        if(len(token)>3 and token.startswith('\u0627') and token[2]=='\u0627'):
+            token = token[0:2] + token[3:]
+            return token
+        if(len(token)>4 and token.startswith('\u0627') and token[-2]=='\u0627'):
+            token = token[1:] 
+            token = token[:-2] + token[-1]
+            return token        
+        
+    def verb(self, token):
+        """ 
+            stem the verb prefixes and suffixes or both
+        """
+        vb = self.verb_t1(token)
+        if vb == None:
+            vb = self.verb_t2(token)
+            if vb == None:
+                vb = self.verb_t3(token)
+                if vb == None:
+                    vb = self.verb_t4(token)
+                    if vb == None:
+                        vb = self.verb_t5(token)
+                        if vb != None:
+                            token = vb
```

Comment:
I think the Pythonic way to code this would use `try...except`, right?

Reply:
Sorry, I am not Pythonist, but I think we cannot use try...except instead of if else statements.

Reply:
Please put all in a `try` block, and have them raise an exception that you catch in your `except` block.

Reply:
Ok thanks, it is fixed now